api/v1alpha1/akodeploymentconfig_types.go

```diff
@@ -341,6 +344,12 @@ type AKORbacConfig struct {
 	PspEnabled *bool `json:"pspEnabled,omitempty"`
 }
 
+// FeatureGates describes the configuration for AKO features
+type FeatureGates struct {
+	// GatewayAPI enables/disables processing of Kubernetes Gateway API CRDs.
+	GatewayAPI string `json:"GatewayAPI,omitempty"`
```

Comment:
Use uppercase `GatewayAPI` here to align with: https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/blob/master/helm/ako/values.yaml#L7